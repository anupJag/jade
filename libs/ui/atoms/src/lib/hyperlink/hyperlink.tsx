import React, { FC, MouseEventHandler } from 'react';
import { Icon } from '@jade/ui/atoms';
import * as S from './hyperlink.styled';

export type LinkProps = {
  href: string;
  type: 'PrimaryNav' | 'SecondaryNav';
  title?: string;
  target?: string;
  hasSubnav?: boolean;
  className?: string;
  onClick?: MouseEventHandler;
};

export const HyperLink: FC<LinkProps> = ({
  href,
  type,
  title,
  target,
  children,
  className,
  hasSubnav = false,
  onClick,
}) => {
  let LinkComp = S.PrimaryNavLink;
  LinkComp = type === 'SecondaryNav' ? S.SecondaryNavLink : LinkComp;
  return (
    <LinkComp className={className} onClick={onClick} href={href} target={target} title={title}>
      {children} {hasSubnav ? <Icon name="chevron-down" aria-hidden="true" /> : null}
    </LinkComp>
  );
};
