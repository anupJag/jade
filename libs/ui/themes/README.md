# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
