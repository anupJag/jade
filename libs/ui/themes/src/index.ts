export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
