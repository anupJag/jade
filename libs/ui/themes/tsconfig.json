{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
