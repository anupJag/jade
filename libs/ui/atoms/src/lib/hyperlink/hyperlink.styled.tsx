import { styled } from '@jade/ui/themes';

export const PrimaryNavLink = styled.a`
  ${props => props.theme.themeColors.navigation.primary}
  position: relative;
  text-decoration: underline;
  &:after {
    width: 100%;
    content: ' ';
    height: ${props => props.theme.space[1]};
    background: ${props => props.theme.themeColors.navigation.afterBackground};
    display: block;
    transform: scaleX(0);
    transition: linear 0.3s transform;
    position: absolute;
    bottom: -10px;
    left: 0;
  }
  &:hover,
  &:focus {
    outline: ${props => props.theme.borders.none};
    text-decoration: none;
    &:after {
      transform: scaleX(1);
    }
  }
`;
export const SecondaryNavLink = styled.a`
  display: flex;
  padding: ${props => props.theme.space[2]};
  border-bottom: ${props => props.theme.borders['1px']}
    ${props => props.theme.themeColors.navigation.secondary.borderBottomColor};
  &:hover,
  &:active,
  &:focus {
    text-decoration: underline;
    outline: ${props => props.theme.borders.none};
  }
`;
