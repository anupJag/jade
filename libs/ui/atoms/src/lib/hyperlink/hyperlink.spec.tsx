import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider } from '@jade/ui/themes';
import { HyperLink } from './hyperlink';

describe('<Link />', () => {
  let LinkComponent;
  beforeEach(() => {
    LinkComponent = render(
      <ThemeProvider>
        {/* With default values */}
        <HyperLink />
        {/* To check conditional change in component */}
        <HyperLink type="SecondaryNav" hasSubnav />
      </ThemeProvider>,
    );
  });

  /**
   * Capturing Sanpshot : Link
   *
   *  To check style / structure changes in Styled Components
   */
  test('snapshot check', () => {
    const { asFragment } = LinkComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
