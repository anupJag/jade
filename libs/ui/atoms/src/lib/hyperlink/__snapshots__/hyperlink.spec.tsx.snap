// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> snapshot check 1`] = `
<DocumentFragment>
  <a
    class="css-mzqm6m"
  >
     
  </a>
  <a
    class="css-1u0qhwh"
  >
     
    <svg
      aria-hidden="true"
      class="css-1im46kq"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;
